controllers/automation/packagedeployment_controller.go

```diff
@@ -203,20 +225,22 @@ metadata:
 		pkgBuf.Nodes = append(pkgBuf.Nodes, nsNode)
 	}
 
-	// If a ClusterScaleProfile CR exists, and one exists for the
-	// associated cluster, then copy the cluster version to the package
-	if scaleProfileIdx > -1 {
-		clusterScaleProfile, err := r.findClusterScaleProfile(ctx, c)
+	// If a binding resource exists in the package, and one exists
+	// that matches the name of the associated cluster, overwrite the
+	// resource in the package with the cluster resource
+
+	for id, pkgResource := range bindingResources {
+		clusterResource, err := r.findClusterObject(ctx, c, id)
 		if err != nil {
-			r.l.Error(err, "error finding cluster scale profile", "cluster", c)
+			r.l.Error(err, "error finding cluster resource", "cluster", c)
```

Comment:
We don't need to raise an error here anymore. With this change, we will allow the objects to not exist; the object inside the package may serve as defaults. Alternatively, a different controller may come and inject the binding later. 

Actually, we might end up wanting to add package conditions for each binding, automatically, to indicate whether they have been bound yet or not. @mortent WDYT of that? If the default is acceptable, the user can manually set the condition to True. If we inject a resource, the binding is satisfied and the condition can be set to True.

Reply:
Pretty sure this is the right thing to do. And @ChrisopherFry can add conditions in the UI, and it will make the UX of this really nice IMO.

Reply:
For now I have just updated this PR so that it doesn't raise an error here anymore. I will take a look at adding conditions. I took a brief look at Morten's demo controller, so can probably do something similar here.